src/domain/relay/limit-addresses.mapper.ts

```diff
@@ -134,16 +134,31 @@ export class LimitAddressesMapper {
       return false;
     }
 
-    // If data of execTransaction is an ERC20 transfer
+    // If data of execTransaction is of ERC20
     try {
       const erc20DecodedData = this.erc20Decoder.decodeFunctionData({
         data: execTransaction.data,
       });
-      // If the ERC20 transfer targets 'self' (the Safe), we consider it to be invalid
-      return (
-        erc20DecodedData.functionName === 'transfer' &&
-        erc20DecodedData.args[0] !== args.to
-      );
+
+      // `transfer` to 'self' (the Safe) is not allowed
+      if (erc20DecodedData.functionName === 'transfer') {
+        const to = erc20DecodedData.args[0];
+        // TODO: Propagate checksummed address types from RelayDto from controller
+        return to !== getAddress(args.to);
+      }
+
+      // `transferFrom` to 'self' or from sender to sender as recipient is not allowed
+      if (erc20DecodedData.functionName === 'transferFrom') {
+        const [sender, recipient] = erc20DecodedData.args;
+        return (
+          sender !== recipient &&
+          // TODO: Propagate checksummed address types from RelayDto from controller
+          recipient !== getAddress(args.to)
+        );
+      }
+
+      // Otherwise allow other methods
+      return true;
```

Comment:
We are returning `true` while only checking the ERC20 related data. We have some further checks that avoid calls targeting "self". Imagine that you have an ERC20 method which is not `transfer` and is no` transferFrom`: shouldn't we validate that the `execTransaction` is not targeting the Safe itself also (the transaction would no succeed if the Safe could not handle that method but we would still relay right?).

Reply:
I simplified the code in dbb5ca98c7f795a1bd0dfb466138f35f85e45f35 to explicitly handle `transfer`/`transferFrom` validation in their own conditions, thus allowing all other ERC-20 methods, providing they are not trying to be called on the Safe. The changes simplify the code, as well as include test cases.